torch/_tensor_docs.py

```diff
@@ -1896,6 +1896,14 @@ def callable(a, b) -> number
 In-place version of :meth:`~Tensor.reciprocal`
 """)
 
+add_docstr_all('record_stream',
+               r"""
+record_stream(stream)
+
+Ensures that the memory of which this tensor is not reused before the given
+stream completes work.
```

Comment:
I think we can improve the doc here a bit. At minimum, if I'm a user, when should I call this function?

Reply:
`completes work` is also a bit unclear. It might be better to say something like ``completes all current work queued on :attr:`stream` ``.

Reply:
Thanks for your review.

I changed my mind. It would be better to warn unexpected memory reuse with more than one stream. I explained it in a note block at the new commit.

```diff
-Ensures that the memory of which this tensor is not reused before the given
-stream completes work.
+Ensures that the tensor memory is not reused for another tensor until all
+current work queued on :attr:`stream` are complete.
+
+.. note::
+
+    The caching allocator is aware of only the stream where a tensor was
+    allocated. Due to the awareness, it already correctly manages the life
+    cycle of tensors on only one stream. But if a tensor is used on a stream
+    different from the stream of origin, the allocator might reuse the memory
+    unexpectedly. Calling this method lets the allocator know which streams
+    have used the tensor.
+
```